File: java/02_library/source/src/main/java/namespace/Generator.java
${{VAR_COPYRIGHT_HEADER}}

${{VAR_NAMESPACE_PACKAGE_DECLARATION}}

/**
 * Dummy interface.
 *
 * @param <T> The type of objects to generate
 *
 */
public interface Generator<T> {

    /**
     * Generates an object of the Generator
     *
     * @return An object of the Generator
     */
    public T generate();

}

File: python/02_library/source/package/generator.py
${{VAR_COPYRIGHT_HEADER}}
"""Module providing a Generator class."""

from abc import ABC, abstractmethod


class Generator(ABC):
    """Dummy abstract base class.

    Attributes:
        basis (str): The base value used by the Generator.
        index (int): The index used by the Generator.
    """

    def __init__(self, basis: str):
        self.basis: str = basis
        self.index: int = 0

    @abstractmethod
    def generate(self) -> str:
        """Generates a dummy string of this Generator.

        Returns:
            str: A dummy string of the Generator
        """
